packages/react-components/src/components/RichTextEditor/RichTextEditor.tsx

```diff
@@ -191,21 +201,47 @@ export const RichTextEditor = React.forwardRef<RichTextEditorComponentRef, RichT
 
   useEffect(() => {
     // don't set callback in plugin constructor to update callback without plugin recreation
-    updatePlugin.onUpdate = (event: string, imageSrcArray?: Array<string>) => {
+    updatePlugin.onUpdate = (
+      event: string,
+      /* @conditional-compile-remove(rich-text-editor-image-upload) */ shouldUpdateInlineImages?: boolean
+    ) => {
       if (editor.current === null) {
         return;
       }
       if (event === UpdateEvent.Blur || event === UpdateEvent.Dispose) {
         onContentModelUpdate && onContentModelUpdate(editor.current.getContentModelCopy('disconnected'));
       } else {
-        onChange && onChange(exportContent(editor.current), imageSrcArray);
+        const content = exportContent(editor.current);
+        /* @conditional-compile-remove(rich-text-editor-image-upload) */
+        let removedInlineImages: Record<string, string>[] = [];
+        /* @conditional-compile-remove(rich-text-editor-image-upload) */
+        if (shouldUpdateInlineImages) {
+          /* @conditional-compile-remove(rich-text-editor-image-upload) */
+          removedInlineImages = getRemovedInlineImages(content, previousInlineImages);
+        }
+
+        onChange &&
+          onChange(content, /* @conditional-compile-remove(rich-text-editor-image-upload) */ removedInlineImages);
+
+        /* @conditional-compile-remove(rich-text-editor-image-upload) */
+        setPreviousInlineImages(getPreviousInlineImages(content));
       }
     };
-  }, [onChange, onContentModelUpdate, updatePlugin]);
+  }, [
+    onChange,
+    onContentModelUpdate,
+    updatePlugin,
+    /* @conditional-compile-remove(rich-text-editor-image-upload) */ previousInlineImages
+  ]);
 
   /* @conditional-compile-remove(rich-text-editor-image-upload) */
   useEffect(() => {
-    copyPastePlugin.onInsertInlineImage = onInsertInlineImage;
+    if (!onInsertInlineImage) {
+      return;
+    }
+    copyPastePlugin.onInsertInlineImage = (imageAttributes: Record<string, string>) => {
+      onInsertInlineImage && onInsertInlineImage(imageAttributes);
```

Comment:
I think this useEffect should looks like this:
```
useEffect(() => {
    copyPastePlugin.onInsertInlineImage = onInsertInlineImage;
}, [copyPastePlugin, onInsertInlineImage]);
```
as it less verbose but do the same thing (mostly because of what is called as a callback in copyPastePlugin (that you check that onInsertInlineImage exists))